pkgs/tools/wayland/swayr/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, fetchFromSourcehut, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "swayr";
+  version = "0.6.0-beta.0";
+
+  src = fetchFromSourcehut {
+    owner = "~tsdh";
+    repo = "swayr";
+    rev = "v${version}";
+    sha256 = "sha256:018qdpl1wabk88rnrlwa2nq5h88hhbh1sv33mcpvwipvhb9q8s90";
+  };
+
+  cargoSha256 = "sha256:0m3cf7i703fkpdr2ap5r5qpwmpcfld745vfwvdf9ll3njmprhj6c";
+
+  # only one check which fails with
+  # thread 'config::test_load_config' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }', src/config.rs:249:14
```

Comment:
It checks for a config dir https://git.sr.ht/~tsdh/swayr/tree/main/item/src/config.rs#L249

Maybe point $HOME to $TMP and you maybe need to create the directory.